src/compiler/checker.ts

```diff
@@ -5255,6 +5245,40 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     )
                 );
             }
+            const exportEqual = symbol.exports.get(InternalSymbolName.ExportEquals);
+            if (exportEqual) {
+                if (exportEqual.flags & SymbolFlags.Alias) {
+                    // copy all symbols from the target alias
+                    const target = resolveSymbol(exportEqual);
+                    const targetExports = getExportsOfSymbol(target);
+                    extendExportSymbols(symbols, targetExports);
+                }
+                else if (exportEqual.declarations) {
+                    loop: for (const node of exportEqual.declarations) {
+                        let init: Expression
+                        switch (node.kind) {
+                            case SyntaxKind.ExportAssignment:
+                                init = (node as ExportAssignment).expression;
+                                break;
+                            case SyntaxKind.BinaryExpression:
+                                init = (node as BinaryExpression).right;
+                                break;
+                            case SyntaxKind.SourceFile:
+                                // JSON source file
+                                break loop;
+                            default:
+                                Debug.failBadSyntaxKind(node, "Unhandled export assignment kind in export= lookup");
+                        }
+                        // copy out the values from the export= target
+                        // NOTE: Must match `resolveStructuredTypeMembers` so identical symbols merge later when other structural elements are copied
+                        const type = getWidenedType(checkExpressionCached(init));
+                        const props = getPropertiesOfType(type);
+                        const table = createSymbolTable(props);
+                        extendExportSymbols(symbols, table);
+                    }
+                }
+                symbols.delete(InternalSymbolName.ExportEquals)

```

Comment:
This maintains the "the resolved exports don't contains an `export=` symbol" expectation most callers have. It's naturally still accessible on the raw, unresolved `symbol.exports` table for callers that need it, though.